tensorbay/opendataset/CACD/loader.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python3
+#
+# Copyright 2021 Graviti. Licensed under MIT License.
+#
+# pylint: disable=invalid-name
+# pylint: disable=missing-module-docstring
+
+import os
+from typing import Any, Dict, Tuple
+
+from ...dataset import Data, Dataset
+from ...exception import ModuleImportError
+from ...label import CategoryInfo, Classification, ClassificationSubcatalog
+from ...utility import NameList
+from .._utility import glob
+
+DATASET_NAME = "CACD"
+_ATTRIBUTES = {
+    "attributes": [
+        {"name": "age", "type": "number", "minimum": 14, "maximum": 62},
+        {"name": "birth", "type": "number", "minimum": 1951, "maximum": 1990},
+        {"name": "range", "type": "number", "minimum": 1, "maximum": 50},
+        {"name": "year", "type": "number", "minimum": 2004, "maximum": 2013},
+        {"name": "name", "type": "string"},
+        {
+            "name": "lfw",
+            "type": "boolean",
+            "description": "Whether the celebrity is in LFW dataset.",
+        },
+    ]
+}
+MAT_KEYS = ["age", "birth", "lfw", "rank", "year"]
+
+
+def CACD(path: str) -> Dataset:
+    """Dataloader of `Cross-Age Celebrity Dataset (CACD)`_ dataset.
+
+    .. _Cross-Age Celebrity Dataset (CACD): https://bcsiriuschen.github.io/CARC/
+
+    The file structure should be like::
+
+        <path>
+            CACD2000/
+                14_Aaron_Johnson_0001.jpg
+                ...
+            celebrity2000.mat
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class:`~tensorbay.dataset.dataset.Dataset` instance.
+
+    """
+    root_path = os.path.abspath(os.path.expanduser(path))
+    dataset = Dataset(DATASET_NAME)
+    dataset.catalog.classification = _get_subcatalog()
+    segment = dataset.create_segment()
+    image_files = glob(os.path.join(root_path, "CACD2000", "*.jpg"))
+    labels_map = _get_labels_map(os.path.join(root_path, "celebrity2000.mat"))
+    for image in image_files:
+        category, attribute = labels_map[image]
+        image_data = Data(os.path.join(root_path, "CACD2000", image))
+        image_data.label.classification = Classification(category, attribute)
+        segment.append(image_data)
+    return dataset
+
+
+def _get_labels_map(path: str) -> Dict[str, Tuple[str, Dict[str, Any]]]:
+    """Get celebrity_image_data from .mat file.
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Raises:
+        ModuleImportError: When the module "h5py" can not be found.
+
+    Returns:
+        A Dict of attributes.
+
+    """
+    try:
+        from h5py import File  # pylint: disable=import-outside-toplevel
+    except ModuleNotFoundError as error:
+        raise ModuleImportError(error.name) from error  # type: ignore[arg-type]
+    mat_file = File(path)
```

Comment:
the deprecation warning from h5py:
```
<ipython-input-61-41418b5a1efe>:1: H5pyDeprecationWarning: The default file mode will change to 'r' (read-only) in h5py 3.0. To suppress this warning, pass the mode you need to h5py.File(), or set the global default h5.get_config().default_file_mode, or set the environment variable H5PY_DEFAULT_READONLY=1. Available modes are: 'r', 'r+', 'w', 'w-'/'x', 'a'. See the docs for details.
```
So you need add mode to `File`: `mat_file = File(path, "r")`